src/main/io/gps.c

```diff
@@ -1337,14 +1341,39 @@ static void calculateNavInterval (void)
     gpsSol.navIntervalMs = constrain(navDeltaTimeMs, 50, 2500);
 }
 
+#if defined(USE_VIRTUAL_GPS)
+static void updateVirtualGPS(void)
+{
+    const uint32_t updateInterval = 100; // 100ms 10Hz update time interval
+    static uint32_t nextUpdateTime = 0;
+    uint32_t currentTime = millis();
+    if (cmp32(currentTime, nextUpdateTime) > 0) {
+        getVirtualGPS(&gpsSol);
+        gpsSol.time = currentTime;     // ms_tow
+        gpsSol.navIntervalMs = updateInterval;
+        gpsSetFixState(GPS_FIX); // fix_type
+        sensorsSet(SENSOR_GPS);
+        nextUpdateTime = currentTime + updateInterval;
+    }
+}
+#endif
+
 void gpsUpdate(timeUs_t currentTimeUs)
 {
     static timeDelta_t gpsStateDurationFractionUs[GPS_STATE_COUNT];
     timeDelta_t executeTimeUs;
     gpsState_e gpsCurrentState = gpsData.state;
     gpsData.now = millis();
 
+#if defined(USE_VIRTUAL_GPS)
+    if (gpsConfig()->provider == GPS_VIRTUAL) {
+        updateVirtualGPS();
+        return;
```

Comment:
Rest of update code (`switch (gpsData.state) {`,  updateGpsIndicator, beeps etc. should be called for GPS_VIRTUAL too. Fix `updateVirtualGPS` or reuse code from https://github.com/betaflight/betaflight/pull/14131/files#diff-15b828e75456a052109384cb9af7da6fb6001ebdf63c417647471ca2201cdee8R1400 if necessary.